### ISSUE TITLE
Fix: Multiple border issues

### ISSUE THREAD

**Issue by dg-ac** [ironcalc/IronCalc#119]
Hi @nhatcher. I've noticed some unexpected behaviours when working with borders:

1. Content disappears after changing border styles. Please check the video below. This doesn't only happen when selecting double borders but with pretty much any change, randomly.

https://github.com/user-attachments/assets/2112e4de-f4b2-4fa1-b2e1-3091913cfc24

2. Cells are not selected when borders panel is open. Can also be seen on the previous video (00:10). Ideally, if such panel is open, clicking on a cell outside it should both close the panel and select such cell.

3. After applying a border style to a cell or range of cells, the button stays active. If after this we select another cell/range of cells and want to apply new border styles, we need to unselect first the old style and then select the new one. Hard to explain, better check the video below.

https://github.com/user-attachments/assets/42c8ccdd-5702-4a08-b6b6-0d15162b738e

4. Applying borders is not always accurate. There are many issues here and they may be related: borders don't disappear when dragging from empty cells; also when we have previously removed borders in another range; etc... It's quicker to understand if you check the video below:

https://github.com/user-attachments/assets/d4444365-a6b1-426d-9a07-f03ddbda9d5e

**Comment by nhatcher** [comment]
1. Fixed
2. Won't fix for now.
This is a tough one. The technical problem is that we can't use Popover for this. I have tried disabling the pointerEvents in the popover. And this mostly works but the Popover  won't close automatically when we click away. We could use React's build in ClickAway but this won't work without substantial reworkings of the events. On a more general note, I wonder home many of this menus/popovers we want to have with this behaviour
3. Fixed
4. Fixed most of the issues
Can you please check? I will open an issue for #2. 
Notice that behaviour may be a bit different than Excel or Google sheets, but I think it is ok

**Comment by dg-ac** [comment]
Thanks for fixing these! I'll answer one by one:
1. LGTM
2. Understood. Maybe, for the time being and as a workaround, we can close the dialog after applying the border? In the same way the 'Fill background' works.
3. LGTM
4. LGTM

**Comment by ivailop7** [comment]
I guess things have changed again, since when trying the latest web app version, none of the border settings are reflected correctly.

**Comment by nhatcher** [comment]
Hi @ivailop7, really? it works for me. Could you tell me what is the problem? What browser/OS are you in?

I think it is not very intuitive how it works at the moment. The default color is white, do you think that is the problem?

Thanks for checking in and reporting!

**Comment by nhatcher** [comment]
I set the default border color to black. Hopefully that fixes the issue

**Comment by ivailop7** [comment]
> I set the default border color to black. Hopefully that fixes the issue

Yeah, that was it actually. My fault for not checking this better. The default was white, so any border style I guess was getting applied but wasn't visible. Thank you for the prompt reply! It's a very cool project!

**Comment by nhatcher** [comment]
> Closing this as 3/4 of the issues were fixed back then. The one remaining may not be an bug, more a preference.
Feel free to open another issue if you think it is worth improving something with the 1/4 remaining.
